src/doc/en/developer/coding_in_python.rst

```diff
@@ -473,6 +473,41 @@ example:
 Note that the syntax in ``except`` is to list all the exceptions that
 are caught as a tuple, followed by an error message.
 
+If you are writing new code to Sage, you soon face the problem of choosing an
+exception most suitable for an errorneous situation. This is often a tricky
+business. The following aims to guide you in choosing from the most relevant
+exceptions to Sage developers. A method (or a function) accepts input described
+in the ``INPUT`` block of :ref:`the docstring <section-docstring-function>` and
+returns output. You may raise
```

Comment:
(Thank you!!!)

I'd suggest to shorten this significantly and slightly rearrange, for example:

* remove the first two sentences (in particular "errorneous situation" is probably misleading)
* move the sentence "A method ... output" to the beginning to the paragraph.  After that, you could continue with "If the input can not be handled by the code, it may raise an exception.", and then "The following...".

Reply:
Good idea. Done.